src/expr.rs

```diff
@@ -549,6 +550,21 @@ pub mod parsing {
         ))
     ));
 
+    named!(expr_loop -> Expr, do_parse!(
+        id: option!(
+            terminated!(
+                preceded!(
+                    punct!("'"),
+                    ident),
+                punct!(":"))) >>
```

Comment:
This should be unindented to be at the same level as "id" above and "loop_body" below.